bpf/process/pfilter.h

```diff
@@ -114,6 +114,162 @@ struct selector_filter {
 	__u32 index;
 };
 
+enum {
+	PF_CRED_UNKNOWN = 0x00, // No filter
+	PF_CRED_RUID = 0x01, // Real UIDs
+	PF_CRED_RGID = 0x02, // Real GIDs
+	PF_CRED_EUID = 0x04, // Effective UIDs
+	PF_CRED_EGID = 0x08, // Effective GIDs
+};
+
+struct cred_id {
+	__u32 start;
+	__u32 end;
+} __attribute__((packed));
+
+struct cred_filter {
+	__u64 ty; /* credentials type flags */
```

Comment:
What are the values of these flags?

Are they used somewhere?
The only thing I'm seeing is:

```
	if (!cred_f->ty)
		return PFILTER_REJECT;
```

so it seems to me that this can be omitted.
If they are used at a later patch, please introduce them there.

Reply:
Good catch, it should be used since we can't predict the order of passed credentials filters uid,euid,x,y or x,uid,y,euid...  in the tracing policy, so every time we iterate to the next one we should check it to know which field we are parsing... I intended to use it for that, will fix it, thanks!